tests/testsuite.src/run_extensions.at

```diff
@@ -5601,3 +5600,143 @@ AT_CHECK([$COBCRUN_DIRECT ./prog], [0],
+          2 Y2-DAT      REDEFINES X-DAT PICTURE LX(5) DEPENDING ON LGY.
+          2 UNUSED-2    PIC X VALUE "!".
+          2 Z-DAT       PICTURE LX(7) DEPENDING ON LGZ VALUE ".......".
+          2 T-DAT       REDEFINES Z-DAT PICTURE LX(7) DEPENDING ON LGT.
+          2 UNUSED-3    PIC X VALUE ".".
+       1 W-LGS.
+          2 LGX         PIC 99 VALUE 10.
+          2 LGY         PIC 9 VALUE 5.
+          2 LGZ         PIC 9 VALUE 3.
+          2 LGT         PIC 9 VALUE 5.
+       PROCEDURE        DIVISION.
+       MAIN.
+           MOVE "0123456789" TO X-DAT
+           DISPLAY X-DAT "*" Y1-DAT "*" Y2-DAT "*"
+           MOVE "0123456" TO Z-DAT
+           DISPLAY T-DAT "*" Z-DAT "*"
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE redefines.cob], [0], [], [])
+AT_CHECK([$COBCRUN_DIRECT ./redefines], [0],
+[0123456789*01234*01234*
+012..*012*
+])
+
+AT_CHECK([$COMPILE -fodoslide redefines.cob], [1], [],
+[redefines.cob:9: error: when ODOSLIDE is set, REDEFINES with PICTURE string with 'L' character are not yet supported
+])
+
+AT_DATA([prog_extraneous_depending.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog_extraneous_depending.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 L		COMP-1.
+       1 F-DAT		PICTURE X(10) DEPENDING ON L.
+       PROCEDURE        DIVISION.
+           MOVE SPACES TO F-DAT
+           STOP RUN.
+])
+AT_CHECK([$COMPILE prog_extraneous_depending.cob], [1], [],
+[prog_extraneous_depending.cob:7: error: DEPENDING clause must only be specified for PICTURE strings with 'L' character
+])
+
+AT_DATA([prog_missing_depending.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog_missing_depending.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 F-DAT		PICTURE LX(10).
+       PROCEDURE        DIVISION.
+           MOVE SPACES TO F-DAT
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE prog_missing_depending.cob], [1], [],
+[prog_missing_depending.cob:6: error: missing DEPENDING clause for PICTURE string with 'L' character
+])
+
+AT_DATA([prog_errs.cob], [
+       IDENTIFICATION   DIVISION.
+       PROGRAM-ID.      prog_errs.
+       DATA             DIVISION.
+       WORKING-STORAGE  SECTION.
+       1 W-LGS.
+         2 LGS		PICTURE X(10).
+       1 W-DATA.
+         2 L-DAT	PICTURE LX(10) DEPENDING ON LGS.
+	 2 L-ERR1       PICTURE L9 DEPENDING ON LGS.
+	 2 L-ERR2       PICTURE XXXXLA(2) DEPENDING ON LGS.
+       PROCEDURE        DIVISION.
+           STOP RUN.
+])
+
+AT_CHECK([$COMPILE prog_errs.cob], [1], [],
+[prog_errs.cob:10: error: DEPENDING clause must only be given for alphabetic or alphanumeric PICTURE strings
+prog_errs.cob:11: error: L must be at start of PICTURE string
+prog_errs.cob:11: error: L cannot follow A or X
+prog_errs.cob:11: error: DEPENDING clause must only be given for alphabetic or alphanumeric PICTURE strings
+])
+
+AT_CLEANUP
```

Comment:
Please add missing runtime tests (per "Data Division - Clauses
47 A2 05UL Rev04 9-35") for:

* depending-on-var <= 1 >= max (not sure what should happen then, what happens with GCOS per default / if runtime checks are enabled/disabled ?)
* `INITIALIZE group-containing-lvar` (should always ignore the size, and likely does already)
* `INITIALIZE lvar` (should always ignore the size, I'm not sure it does)
* MOVE CORRESPONDING when l-var is a sending field (should be only copied from with its current size)
* MOVE CORRESPONDING when l-var is a receiving field (should be copied to with its max size)

Reply:
Thanks for noting the missing tests. Dealing with `INITIALIZE` and `MOVE CORRESPONDING` was tricky ;-)
I am still having some difficulties grasping what `flag_relative_odo` is meant for…

Reply:
I don't find any `relative_odo` in the sources.

Reply:
Yes I cannot perform the runtime checks for now, so that's postponed.

Reply:
I guess those checks were finished now?

Reply:
Yes unless we have missed some instructions. The tests for `PICTURE L` currently included in `run_extensions.at` are as expected on a GCOS 7 platform.